modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/index/hash/InsertHashIndexRowInvokeClosure.java

```diff
@@ -61,7 +67,10 @@ public void call(@Nullable HashIndexRow oldRow) throws IgniteInternalCheckedExce
             return;
         }
 
-        freeList.insertDataRow(hashIndexRow.indexColumns());
+        // If the index columns cannot be fully inlined, then we will write it to the FreeList.
+        if (hashIndexRow.indexColumns().valueSize() > inlineSize + PARTITIONLESS_LINK_SIZE_BYTES) {
```

Comment:
Isn't there a helper method for it somewhere? I don't want this comparison code to be repeated in several places

Reply:
Fix it